docs/generator/checklinks.sh

```diff
@@ -319,9 +313,9 @@ if [ -z "${file}" ] ; then
 		exit 1
 	fi
 	cd ${DOCS_DIR}
-	for f in $(find . -type d \( -path ./${GENERATOR_DIR} -o -path ./node_modules \) -prune -o -name "*.md" -print); do
+	while IFS= read -r -d '' f; do
 		checklinks "$f"
-	done
+	done < <(find . -type d \( -path ./${GENERATOR_DIR} -o -path ./node_modules \) -prune -o -name "*.md" -print)
```

Comment:
I don't see this working in my Ubuntu. I put a similar loop in a while loop with an `echo` instead of a `checklinks` and it just doesn't output anything. I have never used this `< <(...)` syntax before, so I don't know what correction to suggest, but it's what's causing the warnings, because I didn't get any with the previous for loop (from the master branch).

Reply:
OK, I suspected it was probably the issue. If that's the case I'll have to go through a bunch of the scripts to change them away from using this.

For reference, this is the shellcheck error that prompted this particular change: https://github.com/koalaman/shellcheck/wiki/SC2044

Reply:
We should use the "while read" form of the loop here. I've never had problems with writing loops this way. @Ferroin This is primarily the reason I bring up POSIX Sh vs. Bash(ism) here (_no I don't want to start a shell war!_); but in my experience you tend to run into far less problems by writing POSIX Shell and noy relying on Bash :)

Reply:
Which `while read` approach though? The one being commented on here is not POSIX compliant, and the other is honestly harder to read and potentially requires manual setup of a tempfile.

We're also already using the above construct in a couple of other places before this change and it's working fine (I think in retrospect the issue here may be the `-d ' '`, but I'm not sure). I'm inclined to leave the cases that were being changed here as `for x in y` loops, as any files that break that construct are probably going to be breaking a lot of other things (and would have filenames that are of questionable use anyway).

Either way, I'm waiting to look further into this until #7662 is fixed, as that's part of what's breaking the Netlify stuff here.